psydac/linalg/stencil.py

```diff
@@ -1448,50 +1450,26 @@ def set_backend(self, backend):
             self._transpose_func = transpose.func
 
-            nrows   = self._tranpose_args.pop('nrows')
-            ncols   = self._tranpose_args.pop('ncols')
-            gpads   = self._tranpose_args.pop('gpads')
-            pads    = self._tranpose_args.pop('pads')
-            dm      = self._tranpose_args.pop('dm') 
-            cm      = self._tranpose_args.pop('cm')
-            ndiags  = self._tranpose_args.pop('ndiags')
-            ndiagsT = self._tranpose_args.pop('ndiagsT')
-            si      = self._tranpose_args.pop('si')
-            sk      = self._tranpose_args.pop('sk')
-            sl      = self._tranpose_args.pop('sl')
-
-            for i in range(len(nrows)):
-                self._tranpose_args['n{i}'.format(i=i+1)] = nrows[i]
-
-            for i in range(len(nrows)):
-                self._tranpose_args['nc{i}'.format(i=i+1)] = ncols[i]
-
-            for i in range(len(nrows)):
-                self._tranpose_args['gp{i}'.format(i=i+1)] = gpads[i]
+            nrows   = self._transpose_args.pop('nrows')
+            ncols   = self._transpose_args.pop('ncols')
+            gpads   = self._transpose_args.pop('gpads')
+            pads    = self._transpose_args.pop('pads')
+            dm      = self._transpose_args.pop('dm')
+            cm      = self._transpose_args.pop('cm')
+            ndiags  = self._transpose_args.pop('ndiags')
+            ndiagsT = self._transpose_args.pop('ndiagsT')
+            si      = self._transpose_args.pop('si')
+            sk      = self._transpose_args.pop('sk')
+            sl      = self._transpose_args.pop('sl')
 
-            for i in range(len(nrows)):
-                self._tranpose_args['p{i}'.format(i=i+1)] = pads[i]
+            args_names = ['n{i}', 'nc{i}', 'gp{i}', 'p{i}', 'dm{i}',\
+                          'cm{i}', 'nd{i}', 'ndT{i}', 'si{i}' , 'sk{i}', 'sl{i}']
 
-            for i in range(len(nrows)):
-                self._tranpose_args['dm{i}'.format(i=i+1)] = dm[i]
+            args_vals  = [nrows, ncols, gpads, pads, dm, cm, ndiags, ndiagsT, si, sk, sl]
 
-            for i in range(len(nrows)):
-                self._tranpose_args['cm{i}'.format(i=i+1)] = cm[i]
-
-            for i in range(len(nrows)):
-                self._tranpose_args['nd{i}'.format(i=i+1)] = ndiags[i]
-
-            for i in range(len(nrows)):
-                self._tranpose_args['ndT{i}'.format(i=i+1)] = ndiagsT[i]
-
-            for i in range(len(nrows)):
-                self._tranpose_args['si{i}'.format(i=i+1)] = si[i]
-
-            for i in range(len(nrows)):
-                self._tranpose_args['sk{i}'.format(i=i+1)] = sk[i]
-
-            for i in range(len(nrows)):
-                self._tranpose_args['sl{i}'.format(i=i+1)] = sl[i]
+            for arg_name, arg_val in zip(args_names, args_vals):
+                for i in range(len(nrows)):
+                    self._transpose_args[arg_name.format(i=i+1)] = arg_val[i]
```

Comment:
Nice! I suggest that you use a dictionary instead of two lists, as it is much easier to modify it later without making a mistake. You can use the contents of `args_names` as keys, as they are all different strings.

Reply:
done